### PR TITLE
Add leader election for ambient waypoint controller

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/ambient/controller/aggregate.go [hunk 1]
```diff
@@ -118,21 +119,26 @@ func (a *Aggregate) clusterAdded(cluster *multicluster.Cluster, stop <-chan stru
...
-		// TODO handle istiodless remote clusters
-		initAutolabel(opts)
-		go func() {
-			if crdclient.WaitForCRD(gvk.KubernetesGateway, opts.Stop) {
-				waypointController := NewWaypointProxyController(opts.Client, opts.ClusterID, opts.WebhookConfig)
-				waypointController.Run(opts.Stop)
-			}
-		}()
+		election := leaderelection.
+			NewLeaderElection(opts.SystemNamespace, "pod name ", leaderelection.NamespaceController, "", opts.Client).
```
Comment (howardjohn) [inline pilot/pkg/ambient/controller/aggregate.go:129]: `"pod name "` shouldn't be a constant but derived from `args.PodName`

Comment (howardjohn) [reply pilot/pkg/ambient/controller/aggregate.go:129]: I think we want `revision` as well

Comment (hzxuzhonghu) [reply pilot/pkg/ambient/controller/aggregate.go:129]: Yes, i haven't done that

**Comment by hzxuzhonghu** [comment]
The test definitely has goroutine leak, i will take a look

**Pull request merged**